react/features/toolbox/middleware.web.ts

```diff
@@ -36,9 +41,33 @@ MiddlewareRegistry.register(store => next => action => {
     case I_AM_VISITOR_MODE:
     case SET_CONFIG: {
         const result = next(action);
-        const toolbarButtons = _getToolbarButtons(store.getState());
+        const { dispatch, getState } = store;
+        const state = getState();
+
+        if (action.type !== I_AM_VISITOR_MODE) {
+            const {
+                customToolbarButtons,
+                buttonsWithNotifyClick,
+                participantMenuButtonsWithNotifyClick,
+                customParticipantMenuButtons
+            } = state['features/base/config'];
```

Comment:
do we have the guarantee here that the extracted props  already made it to store in case they are part of the UPDATE_CONFIG or OVERWRITE_CONFIG payload? Shouldn't we consider the action's payload here rather than the store values, or a combination of both?

Reply:
What I see is that we merge or override the config in the reducer for these actions. So getting the config value after the action is executed is safe.